### ISSUE TITLE
Enable Configurable Thresholds for Passive Service State

### ISSUE THREAD

**Issue by corygstuart** [NagiosEnterprises/nagioscore#716]
Might it be possible to add functionality to define hard/soft state thresholds for passive checks based on time?  For example, I may not want the service to be in a hard critical state until it's been failing for n minutes.  Until that threshold is reached, it can be in a soft critical state.

**Comment by ericloyd** [comment]
Since passive checks don't know when the next check is run, I think they ignore the normal retry timeline and go straight to a hard state on error, as you've said.  This makes sense since Nagios is not in control over the retry_interval.

I don't have the brainpower right now to fully figure this out, but you could do this with a dependent service that checks the passive service and therefore follows the normal hard/soft retry timeline.  Use that second service to do your notifications and turn them off for the passive check.

**Comment by corygstuart** [comment]
I see what you're saying.  Another check to keep track of the Duration.

**Comment by ericloyd** [comment]
Yes.  Might be a good case for on-demand macros.  You can read about them here (https://assets.nagios.com/downloads/nagioscore/docs/nagioscore/4/en/macros.html).  I'm thinking something like this:

Passive check is called "Check Something Passive" running on "HOST-ABC."  The second check is called "Check the Passive Check" and also runs on HOST-ABC.  "Check something passive" just listens for the passive check and does its thing.  Let's say it expects results every 15 minutes.  Maybe you have a freshness timer on it and a check_dummy result that forces it to go UNKNOWN if it doesn't get a result after 30 minutes; this part is optional.

"Check the Passive Check" then executes the "check_dummy" command with ARG1 set to $SERVICESTATEID:HOST-ABC:Check Something Passive$ on-demand macro.  This way, "Check the Passive Check" is effectively returning the current state of "Check Something Passive."  You can then use normal retry_interval and max_check_attempts to set up your timeline for when "Check the Passive Check" does its notifications.

**Comment by sawolf** [comment]
+1 to @ericloyd's suggestion, that seems like the most reasonable/least labor intensive way to achieve this with the application as it is.

That said, I can definitely see the value in adding support for "retries" or similar in passive check results. No timeline for now but I like the idea.
